File: LENGUAJES/PHP/Ejercicio4.php
<?php
//CICLO FOR

function obtenerFloat($mensaje) {
    while (true) {
        echo $mensaje;
        $input = trim(fgets(STDIN));
        
        if (is_numeric($input) && $input >= 0) {
            return floatval($input);
        } else {
            echo "Error, por favor ingrese un número válido y positivo.\n";
        }
    }
}

function obtenerInt($mensaje) {
    while (true) {
        echo $mensaje;
        $input = trim(fgets(STDIN));
        
        if (is_numeric($input) && intval($input) == $input && $input >= 0) {
            return intval($input);
        } else {
            echo "Error, por favor ingrese un número entero positivo.\n";
        }
    }
}

function inversion() {
    $cant = obtenerFloat("Ingrese la cantidad a invertir: $");
    $inter = obtenerFloat("Ingrese el interés anual (en %): ");
    $años = obtenerInt("Ingrese el número de años: ");
    
    echo "\nResumen de la inversión:\n";
    for ($i = 1; $i <= $años; $i++) {
        $cant += $cant * ($inter / 100);
        echo "Año $i: Capital obtenido será: $" . number_format($cant, 2) . "\n";
    }
}
inversion();
?>

File: LENGUAJES/PYTHON/Ejercicio4.py
#CICLO FOR

def obtener_float(mensaje):
    while True:
        try:
            valor = float(input(mensaje))
            if valor < 0:
                print("El valor debe ser un número positivo. Inténtalo de nuevo.")
            else:
                return valor
        except ValueError:
            print("Error, por favor ingrese un número válido.")

def obtener_int(mensaje):
    while True:
        try:
            valor = int(input(mensaje))
            if valor < 0:
                print("El valor debe ser un número entero positivo. Inténtalo de nuevo.")
            else:
                return valor
        except ValueError:
            print("Error, por favor ingrese un número válido.")

def inversion():
    cant = obtener_float("Ingrese la cantidad a invertir: $")
    inter = obtener_float("Ingrese el interés anual (en %): ")
    años = obtener_int("Ingrese el número de años: ")
    
    print("\nResumen de la inversión:")
    for i in range(1, años + 1):
        cant += cant * (inter / 100)
        print(f"Año {i}: Capital obtenido será: ${cant:.2f}")       
inversion()